### PR TITLE
LMS-1690 moodle-logstore_xapi: add replay multiple events.

### PR DESCRIPTION
**Pull request by lzabo**
**Description**
- LMS-1690 Replay multiple events.
- LMS-1691 Create confirmation modal.

**Related Issues**
No issue

**PR Type**
- Enhancement

### CODE REVIEW

File: lang/en/logstore_xapi.php [hunk 1]
```diff
@@ -121,3 +121,8 @@
 $string['failed'] = 'Failed';
 $string['resendevents'] = 'Resend ({$a->count})';
 $string['replayevent'] = 'Replay event';
+$string['confirmresendeventsheader'] = 'Set to replay events';
+$string['confirmresendevents'] = 'Warning: You are about send events to be replayed.
+ The events what you selected ({$a->count}) will be set to replay and be removed from the error report. 
+ The next running scheduled task will replay this elements.
+ Do you want to continue to do that?';
```
Comment (stephen-oh) [inline lang/en/logstore_xapi.php]: This section needs slightly reworded but Gordon has said he is doing this now

**Review by stephen-oh — COMMENTED** [review]
Once the string has been updated I'll be happy to approve

File: lang/en/logstore_xapi.php [hunk 2]
```diff
@@ -121,3 +121,8 @@
 $string['failed'] = 'Failed';
 $string['resendevents'] = 'Resend ({$a->count})';
 $string['replayevent'] = 'Replay event';
+$string['confirmresendeventsheader'] = 'Set to replay events';
+$string['confirmresendevents'] = 'Warning: You are about send events to be replayed.
```
Comment (gordonmacqueen-lp) [inline lang/en/logstore_xapi.php]: I think the message needs to be shorter
https://uxplanet.org/5-essential-ux-rules-for-dialog-design-4de258c22116

What do you all think of something like:
```
You are about to send {$a->count} record(s) to the queue for reprocessing.
Do you wish to continue?
```
@Patches- @mlynn-lp @lzabo ?

**Review by gordonmacqueen-lp — CHANGES_REQUESTED** [review]
I've just noticed a few typos in the javascript from a previous PR:
- `.reply-event`
- `this.addReplyEvents();`

I assume these should be 'replay', but let's leave it as is for now. We can tidy up the naming later if we need to

File: lang/en/logstore_xapi.php [hunk 3]
```diff
@@ -121,3 +121,8 @@
 $string['failed'] = 'Failed';
 $string['resendevents'] = 'Resend ({$a->count})';
 $string['replayevent'] = 'Replay event';
+$string['confirmresendeventsheader'] = 'Set to replay events';
```
[the remark this answers is not in the document]
Comment (lzabo) [reply lang/en/logstore_xapi.php]: Nope this is definitely not.

This one is the header of the confirmation window.

File: lang/en/logstore_xapi.php (hunk 2 above)
Comment (lzabo) [reply lang/en/logstore_xapi.php]: I agree.

Comment (mlynn-lp) [reply lang/en/logstore_xapi.php]: Could the "Do you wish to continue?" part of the message be shown on a new line?

File: lang/en/logstore_xapi.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (gordonmacqueen-lp) [reply lang/en/logstore_xapi.php]: In that case, how about 'Resend Events'?

File: lang/en/logstore_xapi.php (hunk 2 above)
Comment (lzabo) [reply lang/en/logstore_xapi.php]: Nope, I can not find any exasmple how I can do that.
Just let me know how.

File: lang/en/logstore_xapi.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (lzabo) [reply lang/en/logstore_xapi.php]: I agree with that, and changed it.

**Review by gordonmacqueen-lp — APPROVED** [review]

**Pull request merged**